symbolic-symcache/src/writer.rs

```diff
@@ -85,15 +86,13 @@ where
         Ok(())
     }
 
-    /// Writes a segment as binary data to the writer and returns the [`Seg`] reference.
+    /// Writes a slice as binary data and returns a [`Seg`](format::Seg) (i.e., a fat pointer) containing
```

Comment:
_nit_: Since this is the summary line, you could keep it should and move more description into the next paragraph(s).